### PR TITLE
add layout mapping API for external plugins

### PR DESCRIPTION
**Pull request by therealdreg**
## Description

This fixes all the issues with the context panes and it now works as it did before the plugins were created. 

In the GEF part, it was just a minor refactor and a new external API. 

In the gef-extras part, it was only necessary to call this new function!

I think the proposed solution is simple and it works.

gef-extras code:
- https://github.com/hugsy/gef-extras/pull/111

## Checklist

-  [X] My code follows the code style of this project.
-  [X] My change includes a change to the documentation, if required.
-  [X] If my change adds new code, [adequate tests](docs/testing.md) have been added.
-  [X] I have read and agree to the **CONTRIBUTING** document.

### CODE REVIEW

File: gef.py [hunk 1]
```diff
@@ -4550,6 +4550,27 @@ def pane_title():
...
+    condition: an optional callback: if not None, the callback will be executed first. If it returns true,
+      then only the pane title and content will displayed. Otherwise, it's simply skipped.
+
+    Example usage for a simple text to show when we hit a syscall:
+    def only_syscall(): return gef_current_instruction(gef.arch.pc).is_syscall()
+    def display_pane():
+      gef_print("Wow, I am a context pane!")
+    def pane_title():
+      return "example:pane"
+    register_external_context_pane("example_pane", display_pane, pane_title, only_syscall)
```
Comment (hugsy) [inline gef.py:4569]: This should go in the docs, not the code

**Review by hugsy — CHANGES_REQUESTED** [review]

File: gef.py (hunk 1 above)
Comment (therealdreg) [reply gef.py]: removed

Comment (hugsy) [reply gef.py]: I was suggesting to move this part to docs not removing it. It still has value IMO, just not in the source code of gef, but more for people using it

Comment (therealdreg) [reply gef.py]: done!

**Comment by therealdreg** [comment]
done! @hugsy take a look!

**Review by hugsy — APPROVED** [review]
Great stuff, super clear

**Pull request merged**